src/replit/database/database.py

```diff
@@ -114,12 +116,22 @@ async def get_raw(self, key: str) -> str:
         Returns:
             str: The value of the key
         """
-        async with self.sess.get(self.db_url + "/" +
-                                 urllib.parse.quote(key)) as response:
-            if response.status == 404:
-                raise KeyError(key)
-            response.raise_for_status()
-            return await response.text()
+        for i in range(self.retry_count):
+            try:
+                async with self.sess.get(self.db_url + "/" +
+                                         urllib.parse.quote(key)) as response:
+                    if response.status == 404:
+                        raise KeyError(key)
+                    response.raise_for_status()
+                    return await response.text()
+            except KeyError:
+                raise
+            except Exception as e:
```

Comment:
here it's better to limit ourselves to network errors _or_ http errors in the range 5xx: anything else should be bubbled up immediately

Reply:
Good point. I replace the 'manual' retries with `aiohttp_retry` (for AsyncDatabase) and added a Retry adapter (for the regular, sync Database). `aiohttp_retry` only retries on 500 errors, and I specified the retry adapter to retry only on `[500, 502, 503, 504]`.